.github/actions/prepare-node/action.yml

```diff
@@ -15,27 +10,20 @@ inputs:
 runs:
   using: composite
   steps:
-    # Log debug information
+    # Setup node version from .`nvmrc, and caching.
+    - uses: actions/setup-node@v2
+      with:
+        node-version-file: '.nvmrc'
+        cache: 'npm'
+
+    # Log debug information.
     - shell: sh -e {0}
       run: |
         node --version
         npm --version
 
-    # Get npm cache directory
-    - uses: actions/cache@v2
-      id: npm-cache
-      with:
-        path: ~/.npm
-        key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-node-
-
-    # Install node dependencies
+    # Install node dependencies.
     - shell: sh -e {0}
-      # Update npm cache directory if package-lock has changed
+      # `actions/setup-node` should update npm cache directory if `package-lock` has changed.
       run: |
-        if [ "${{ inputs.install-deps }}" = "yes" ] || [ "${{ steps.npm-cache.outputs.cache-hit }}" != "true" ]; then
```

Comment:
> The install-deps option with "maybe" value was used to handle updating the ~/.npm cache by running npm ci to pre-download the package when the cache is not hit, so that subsequent jobs can directly skip the cache updates and save some time.

I thought, that's what `actions/setup-node@v2`'s  `[`cache: 'npm'` is for](https://github.com/woocommerce/google-listings-and-ads/pull/1133/files#diff-bbf9b957d0e2e4f07bdbe3c523584aea587cd63068c0f45c69c1a8fb6120dcf9R17)

But it seems even though, the cache is restored, it still runs install https://github.com/woocommerce/google-listings-and-ads/runs/4396890740?check_suite_focus=true#step:3:24

I'll try using `cache-hit` explicitly

Reply:
I tried using `cache-hit` to avoid doing `npm ci` but then it seems the node_modules is not there 
https://github.com/woocommerce/google-listings-and-ads/runs/4407826045?check_suite_focus=true#step:3:24

Reply:
Ah, now I got it `actions/setup-node@v2` caches the `~/.npm/` cache only, not `node_modules`. So we do need to call `npm ci` if the cache is hit. I'd revert my experiments, to what was here  in OP.

When we want, to improve that we can do that in a new PR, and as you said, either 
- _"remove all "Setup for jobs" to see if the overall time consumed make no difference, or even less."_
- or make it so subsequent steps (CSS, JS linting) would not call `npm ci` at all, to make thing simple